crates/sp-domains-fraud-proof/src/lib.rs

```diff
@@ -102,6 +103,8 @@ pub struct DomainInherentExtrinsicData {
     pub consensus_transaction_byte_fee: Balance,
     pub domain_chain_allowlist: DomainAllowlistUpdates,
     pub maybe_sudo_runtime_call: Option<Vec<u8>>,
+    pub maybe_evm_domain_contract_creation_allowed_by_call:
```

Comment:
This changes both the input and output of the host function, while both the input and output will SCALE encoded and if this value is `None` it should be fine, but I think it is better to have test to ensure it won't break anything.

Reply:
> Tests are coming in the next PR, can you point me to an example of what this test looks like?

I'm actually not comfortable pushing this without testing. Reason being any runtime upgrades before the tests and might need potential fix will break the fraud proofs

Reply:
>can you point me to an example of what this test looks like?

It is hard to do that in code because a lot of infra will be needed, but you can set up a local dev network with an old release and then upgrade the node with binary built from this branch